lib/dpl/provider/scalingo.rb

```diff
@@ -6,57 +6,28 @@
+        unless context.shell 'curl -OL https://cli-dl.scalingo.io/release/scalingo_latest_linux_amd64.tar.gz && tar -zxvf scalingo_latest_linux_amd64.tar.gz && mv scalingo_*_linux_amd64/scalingo . && rm scalingo_latest_linux_amd64.tar.gz && rm -r scalingo_*_linux_amd64'
           error "Couldn't install Scalingo CLI."
         end
       end
 
       def initialize(context, options)
         super
+        raise Error, 'you must specify your app name with `--app`' if options[:app] && options[:app].empty?
+
         @options = options
-        @remote = options[:remote] || "scalingo"
-        @branch = options[:branch] || "master"
+        @remote = "git@scalingo.com:#{options[:app]}.git"
+        @branch = options[:branch] || 'master'
       end
 
       def logged_in
-        context.shell "DISABLE_INTERACTIVE=true ./scalingo login 2> /dev/null > /dev/null"
+        context.shell 'DISABLE_INTERACTIVE=true ./scalingo login 2> /dev/null > /dev/null'
       end
 
       def check_auth
         if @options[:api_key]
-          unless context.shell "mkdir -p ~/.config/scalingo"
-            error "Couldn't create authentication file."
-          end
-          url = URI.parse('http://api.scalingo.com/v1/users/self')
-          http = Net::HTTP.new(url.host, url.port)
-          request = Net::HTTP::Get.new(url.request_uri)
-          request.basic_auth("", @options[:api_key])
-          request["Accept"] = "application/json"
-          request["Content-type"] = "application/json"
-          response = http.request(request)
-          data = {}
-          if File.exist?("#{Dir.home}/.config/scalingo/auth")
-            data = JSON.parse(File.read("#{Dir.home}/.config/scalingo/auth"))
-          end
-          begin
-            user = JSON.parse(response.body)
-          rescue
-            error "Invalid API token."
-          end
-          data["auth_config_data"] = {}
-          data["auth_config_data"]["api.scalingo.com"] = {}
-          data["auth_config_data"]["api.scalingo.com"]["id"] = user["user"]["id"]
-          data["auth_config_data"]["api.scalingo.com"]["last_name"] = user["user"]["last_name"]
-          data["auth_config_data"]["api.scalingo.com"]["username"] = user["user"]["username"]
-          data["auth_config_data"]["api.scalingo.com"]["email"] = user["user"]["email"]
-          data["auth_config_data"]["api.scalingo.com"]["first_name"] = user["user"]["first_name"]
-          data["auth_config_data"]["api.scalingo.com"]["auth_token"] = @options[:api_key]
-          data["last_update"] = DateTime.now
-          f = File.open("#{Dir.home}/.config/scalingo/auth", "w+") {
-            |f| f.write(data.to_json)
-          }
+          context.shell "timeout 2 ./scalingo login --api-key #{@options[:api_key]} 2> /dev/null > /dev/null"
```

Comment:
Is this compaction really desirable? It is also hiding STDOUT and STDERR without any chance of troubleshooting. I feel that this is wrong.